android/src/main/java/app/notifee/core/ForegroundService.java

```diff
@@ -45,10 +46,14 @@ static void start(int hashCode, Notification notification, Bundle notificationBu
     intent.putExtra("notification", notification);
     intent.putExtra("notificationBundle", notificationBundle);
 
-    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+    // TODO pass this config from notificationBundle
+    boolean startServiceWithPromise = false;
+
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O && startServiceWithPromise) {
+      Log.d(TAG, "Using startForegroundService");
       ContextHolder.getApplicationContext().startForegroundService(intent);
     } else {
-      // TODO test this on older device
+      Log.d(TAG, "Using startService");
```

Comment:
@mieszko4 quick question — is this allowed on Android 12 or do we need to use `startForegroundService` for Android 12 and above?

Reply:
I tested on Android 12 - it was all working well.